pkg/sources/postman/postman_client.go

```diff
@@ -315,6 +305,55 @@ func (c *Client) getPostmanResponseBodyBytes(ctx trContext.Context, urlString st
 	return body, nil
 }
 
+// handleRateLimits processes the rate limit headers from the Postman API response
+// and updates the client's metrics accordingly. If the monthly rate limit usage exceeds
+// the set threshold, an error is returned to abort further processing.
+func (c *Client) handleRateLimits(ctx trContext.Context, resp *http.Response) error {
+	rateLimitRemainingMonthValue := resp.Header.Get("RateLimit-Remaining-Month")
+	if rateLimitRemainingMonthValue == "" {
+		rateLimitRemainingMonthValue = resp.Header.Get("X-RateLimit-Remaining-Month")
+	}
+	if rateLimitRemainingMonthValue == "" {
+		ctx.Logger().V(2).Info("RateLimit-Remaining-Month header not found in response")
+		return nil
+	}
+
+	rateLimitRemainingMonth, err := strconv.Atoi(rateLimitRemainingMonthValue)
+	if err != nil {
+		ctx.Logger().Error(err, "Couldn't convert RateLimit-Remaining-Month to an int",
+			"header_value", rateLimitRemainingMonthValue,
+		)
+		return nil
+	}
+	c.Metrics.apiMonthlyRequestsRemaining.WithLabelValues().Set(
+		float64(rateLimitRemainingMonth),
+	)
+
+	rateLimitTotalMonthValue := resp.Header.Get("RateLimit-Limit-Month")
+	if rateLimitTotalMonthValue == "" {
+		rateLimitTotalMonthValue = resp.Header.Get("X-RateLimit-Limit-Month")
+	}
+	if rateLimitTotalMonthValue == "" {
+		ctx.Logger().V(2).Info("RateLimit-Limit-Month header not found in response")
+		return nil
+	}
+	rateLimitTotalMonth, err := strconv.Atoi(rateLimitTotalMonthValue)
+	if err != nil {
```

Comment:
This is good but one question: can `rateLimitTotalMonth` ever be zero?
If yes, maybe add a zero check because I hate _divided by zero errors_.